lib/Conversion/TorchToLinalg/Utils.cpp

```diff
@@ -67,6 +68,51 @@ Value torch_to_linalg::getZeroPaddedTensor(
   return getPaddedTensor(op, b, input, paddingInts, paddingInts, c0);
 }
 
+// Helper function that adds dynamic padding to a tensor, ignoring unpaddedDims
+// dimensions at the beginning. The high and low padding are the same, and the
+// padding value is zero.
+Value torch_to_linalg::getDynamicZeroPaddedTensor(
+    Operation *op, OpBuilder &b, Value &input, SmallVectorImpl<Value> &padding,
+    int unpaddedDims) {
+  assert(input.getType().isa<RankedTensorType>() &&
+         "input must be RankedTensorType");
+  Location loc = op->getLoc();
+
+  Value c1 = b.create<arith::ConstantOp>(loc, b.getIndexAttr(1));
+  SmallVector<Value> inputDims = getTensorSizes(b, loc, input);
+
+  Value c0 = b.create<arith::ConstantOp>(loc, b.getI64IntegerAttr(0));
+  SmallVector<Value> paddingIncludingUnchanged{c0, c0};
+  paddingIncludingUnchanged.append(padding);
+  for (auto pad = paddingIncludingUnchanged.begin();
+       pad < paddingIncludingUnchanged.end(); pad++)
+    *pad = castIntToIndex(b, loc, *pad);
+
+  Value c2 = b.create<arith::ConstantOp>(loc, b.getI64IntegerAttr(2));
+  auto inputRank = input.getType().cast<RankedTensorType>().getRank();
+  SmallVector<Value> outputDims{inputDims[0], inputDims[1]};
+  for (auto i = 0; i < inputRank; i++) {
+    if (i >= unpaddedDims) {
+      Value padDouble = b.create<arith::MulIOp>(loc, padding[i - 2], c2);
+      Value dim = b.create<arith::AddIOp>(loc, inputDims[i],
+                                          castIntToIndex(b, loc, padDouble));
+      outputDims.push_back(dim);
+    }
+  }
+
+  Type elementType = input.getType().cast<RankedTensorType>().getElementType();
+  Type inputType = RankedTensorType::get(
+      llvm::ArrayRef<int64_t>(SmallVector<int64_t>(inputRank, kUnknownSize)),
+      elementType);
+  input = b.create<tensor::CastOp>(loc, inputType, input);
+
+  SmallVector<Value> strides(inputRank, c1);
+  Value cf0 =
+      b.create<arith::ConstantOp>(loc, b.getFloatAttr(elementType, 0.0));
+  return createPadScalarOp(input, cf0, paddingIncludingUnchanged,
```

Comment:
Any reason why we can't use the existing helper `tensor::createPadScalarOp` ?

Reply:
Yep, the existing help requires `OpFoldResult`s, which as far as I'm aware have to be static values. This should probably get upstreamed, but until it does putting it here is the only solution I could think of.

Reply:
You can convert dynamic values to OpFoldResult with this helper https://github.com/llvm/llvm-project/blob/e1318078a4e160eb723bcbcfcdcc9a1b618f7067/mlir/include/mlir/Dialect/Utils/StaticValueUtils.h#L51